src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,700 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance  the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "window-covering-server.h"
+
+#include <app/util/af.h>
+
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+
+#include <app/CommandHandler.h>
+#include <app/common/gen/attributes/Accessors.h>
+#include <app/reporting/reporting.h>
+#include <app/util/af-event.h>
+#include <app/util/af-types.h>
+#include <app/util/attribute-storage.h>
+#include <string.h>
+
+#ifdef EMBER_AF_PLUGIN_SCENES
+#include <app/clusters/scenes/scenes.h>
+#endif // EMBER_AF_PLUGIN_SCENES
+
+using namespace chip::app::Clusters;
+
+#define CHECK_BOUNDS_INVALID(MIN, VAL, MAX) ((VAL < MIN) || (VAL > MAX))
+#define CHECK_BOUNDS_VALID(MIN, VAL, MAX) (!CHECK_BOUNDS_INVALID(MIN, VAL, MAX))
+
+#define WC_PERCENTAGE_COEF 100
+#define WC_PERCENT100THS_MAX 10000
+
+static EmberAfStatus writeAttribute(chip::EndpointId ep, chip::AttributeId attributeID, uint8_t * dataPtr,
+                                    EmberAfAttributeType dataType)
+{
+    if (!dataPtr)
+        return EMBER_ZCL_STATUS_INVALID_FIELD;
+
+    EmberAfStatus status =
+        emberAfWriteAttribute(ep, ZCL_WINDOW_COVERING_CLUSTER_ID, attributeID, CLUSTER_MASK_SERVER, dataPtr, dataType);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfWindowCoveringClusterPrint("Err: WC Writing Attribute failed, ep:%u, attr:%04x, err:%04x", ep, attributeID, status);
+    }
+    return status;
+}
```

Comment:
What was the reason to revert back to the old API?

Reply:
Ricardo roll back to the old api since it is buggy 
seems on trial that this new Api doesn't change the Attributes properly